File: connectors/connector-basecamp-classic/src/main/java/com/taskadapter/connector/basecamp/classic/BasecampConfigValidator.java
package com.taskadapter.connector.basecamp.classic;

import com.taskadapter.connector.basecamp.FieldNotSetException;
import com.taskadapter.connector.definition.WebConnectorSetup;

public class BasecampConfigValidator {
    public static void validateServerAuth(WebConnectorSetup setup) throws FieldNotSetException {
        final String apiKey = setup.apiKey();
        if (apiKey == null || apiKey.isEmpty()) {
            throw new FieldNotSetException("auth");
        }
        final String apiUrl = setup.host();
        if (apiUrl == null || apiUrl.isEmpty()) {
            throw new FieldNotSetException("api-url");
        }

    }

    public static void validateProjectKey(BasecampClassicConfig config)
            throws FieldNotSetException {
        final String pKey = config.getProjectKey();
        if (pKey == null || pKey.isEmpty()) {
            throw new FieldNotSetException("project-key");
        }
    }

    public static void validateTodoList(BasecampClassicConfig config)
            throws FieldNotSetException {
        final String pKey = config.getTodoKey();
        if (pKey == null || pKey.isEmpty()) {
            throw new FieldNotSetException("todo-key");
        }
    }

}

File: connectors/connector-basecamp-classic/src/main/scala/com/taskadapter/connector/basecamp/classic/BasecampConfigValidator.scala
package com.taskadapter.connector.basecamp.classic

import com.taskadapter.connector.basecamp.FieldNotSetException
import com.taskadapter.connector.definition.WebConnectorSetup
import com.taskadapter.connector.definition.exceptions.BadConfigException

import scala.collection.{Seq, mutable}

object BasecampConfigValidator {
  def validateServerAuth(setup: WebConnectorSetup): Unit = {
    failIfErrors(validateServerAuthNoException(setup))
  }

  def validateServerAuthNoException(setup: WebConnectorSetup): Seq[BadConfigException] = {
    val seq = new mutable.ListBuffer[BadConfigException]()
    val apiKey = setup.getApiKey
    if (apiKey == null || apiKey.isEmpty) seq += FieldNotSetException("auth")
    val apiUrl = setup.getHost
    if (apiUrl == null || apiUrl.isEmpty) seq += FieldNotSetException("api-url")
    seq
  }

  def validateProjectKey(config: BasecampClassicConfig): Unit = {
    failIfErrors(validateProjectKeyNoException(config))
  }

  def validateProjectKeyNoException(config: BasecampClassicConfig): Seq[BadConfigException] = {
    val seq = new mutable.ListBuffer[BadConfigException]()
    val pKey = config.getProjectKey
    if (pKey == null || pKey.isEmpty) seq += FieldNotSetException("project-key")
    seq
  }

  def validateTodoList(config: BasecampClassicConfig): Unit = {
    failIfErrors(validateTodoListNoException(config))
  }

  def validateTodoListNoException(config: BasecampClassicConfig): Seq[BadConfigException] = {
    val seq = new mutable.ListBuffer[BadConfigException]()
    val pKey = config.getTodoKey
    if (pKey == null || pKey.isEmpty) seq += FieldNotSetException("todo-key")
    seq
  }

  @throws[BadConfigException]
  private def failIfErrors(errors: Seq[BadConfigException]): Unit = {
    if (errors.nonEmpty) throw errors.head
  }
}